### PR TITLE
Updated dependancies with latest node modules

### PR DESCRIPTION
**Pull request by karnan4ever**
Signed-off-by: Karnan <kchidamb@redhat.com>

### CODE REVIEW

**Comment by karnan4ever** [comment]
@kamleshverma1 @debloper @gnehapk

**Comment by karnan4ever** [comment]
@TimothyAsir

File: Makefile [hunk 1]
```diff
@@ -11,27 +11,30 @@ DISTDIR   := $(TARDIR)/dist
 TARNAME   := $(TARDIR).tar.gz
 APP_DIST  := ./dist
 
-build-all: build-setup build
+build-all: setup compile tar
```
Comment (debloper) [inline Makefile]: Can we create the wrapper params as `build-tar` and `build-rpm`?
This is pretty easy to do, with the following code, by just maintaining the procedural flow.

File: Makefile [hunk 2]
```diff
@@ -11,27 +11,30 @@ DISTDIR   := $(TARDIR)/dist
...
+compile:
 	gulp release
 
-dist: build
+tar: compile
 	@echo "making dist tarball in $(TARNAME)"
 	mkdir -p $(DISTDIR)
-	cp README.md $(TARDIR)/.
+	mkdir -p $(TARDIR)/docs
+	cp ./docs/* $(TARDIR)/docs/.
```
Comment (debloper) [inline Makefile]: `cp -r ...`

File: Makefile [hunk 3]
```diff
@@ -11,27 +11,30 @@ DISTDIR   := $(TARDIR)/dist
...
 	npm install
 
-build:
+build: compile
+
+compile:
 	gulp release
 
-dist: build
+tar: compile
```
Comment (debloper) [inline Makefile]: Needs a little tweak here, to abstract itself for `build-tar`

File: Makefile [hunk 4]
```diff
@@ -11,27 +11,30 @@ DISTDIR   := $(TARDIR)/dist
...
+	cp ./docs/* $(TARDIR)/docs/.
 	cp -r $(APP_DIST)/* $(DISTDIR)
 	tar -zcf $(TARDIR).tar.gz $(TARDIR);
 	rm -rf $(TARDIR)
 	@echo "--------------------------------------------------"
 	@echo "TAR file is now available at: $(TARNAME)"
 	@echo "--------------------------------------------------"
 
-rpm: dist
+rpm: tar
```
Comment (debloper) [inline Makefile]: Needs a little tweak here, to abstract itself for `build-rpm`

File: gulpfile.js [hunk 5]
```diff
@@ -233,50 +229,17 @@ gulp.task("ut", function (done) {
...
 gulp.task("dev", ["common", "watcher"], function (done) {
     gutil.log(colors.bold.yellow("Watchers Established. You can now start coding"));
-    runSequence("upload", done);
 });
 
 // production mode task
-gulp.task("release", ["common", "compress"], function (done) {
-    runSequence("ut", "upload", done);
+gulp.task("release", ["common"], function (done) {
+    runSequence("ut", done);
```
Comment (debloper) [inline gulpfile.js]: What was `"ut"`?
Needs to be a little verbose, imo.

File: tendrl-frontend.spec [hunk 6]
```diff
@@ -33,9 +33,8 @@ echo "clean"
...
 %files
 %{_datadir}/Tendrl/webapp/*
-%doc docs/*
+%doc ./docs/*
 
 %changelog
-* Ddd Mmm DD YYYY <email@domain.tld> x.y.z-c
-- Fix 1
-- Fix 2
+* Thu Nov 10 2016 kchidamb@redhat.com 0.0.1-1
```
Comment (debloper) [inline tendrl-frontend.spec]: Email actually has to contain the `<...>` angle brackets.

File: gulpfile.js (hunk 5 above)
Comment (gnehapk) [inline gulpfile.js]: ut stands for unit-testing

**Comment by debloper** [comment]
I think I'll have to patch the makefile afterwards; I think @karnan4ever misunderstood my first suggestion.

My suggestion was, instead of using `build-all` (which then again needs tar and rpm building step) we directly use `tar` and `rpm` as build steps. So that it's one step process for almost all task.

```bash
$ make dist    # builds from the source
$ make tar     # compiles, then creates tarball
$ make rpm     # compiles, creates tarball, then prepares rpm
```

**Pull request merged**